src/v/cluster/health_monitor_types.h

```diff
@@ -100,10 +102,18 @@ struct topic_status
   : serde::envelope<topic_status, serde::version<0>, serde::compat_version<0>> {
     static constexpr int8_t current_version = 0;
 
+    topic_status() = default;
+    topic_status(model::topic_namespace, ss::chunked_fifo<partition_status>);
+    topic_status& operator=(const topic_status&);
+    topic_status(const topic_status&);
+    topic_status& operator=(topic_status&&) = default;
+    topic_status(topic_status&&) = default;
+    ~topic_status() = default;
```

Comment:
It's sort of nice that ss::chunked_fifo doesn't support copying by default. I added it for this method because as far as I can tell `ss::map_reduce0` needs copyable objects (used in `health_monitor_backend::collect_topic_status`)